source/extensions/transport_sockets/tls/ssl_handshaker.cc

```diff
@@ -159,6 +159,7 @@ const std::string& SslHandshakerImpl::urlEncodedPemEncodedPeerCertificateChain()
         absl::StrReplaceAll(
             pem, {{"\n", "%0A"}, {" ", "%20"}, {"+", "%2B"}, {"/", "%2F"}, {"=", "%3D"}}));
   }
+  sk_X509_pop_free(cert_chain, X509_free);
```

Comment:
is this fixing a leak? is this present upstream too or it's our addition?

Reply:
I haven't checked, but it's probably specific to use. Pretty sure that upstream this  would have been caught it in CI runs